EvmYul/EVM/Semantics.lean

```diff
@@ -1086,14 +1083,49 @@ def Θ (debugMode : Bool)
       weiValue  := v' -- Equation (137)
       depth     := e  -- Equation (138)
       perm      := w  -- Equation (139)
-      code      := c  -- Note that we don't use an address, but the actual code. Equation (141)-ish.
+      -- Note that we don't use an address, but the actual code. Equation (141)-ish.
+      code      :=
+        match c with
+          | ToExecute.Precompiled _ => default
+          | ToExecute.Code code => code
       header    := H
     }
 
-
   -- Equation (131)
   -- Note that the `c` used here is the actual code, not the address. TODO - Handle precompiled contracts.
-  let (createdAccounts, σ'', g'', A'', out) ← Ξ debugMode fuel createdAccounts σ₁ g A I
+  let (createdAccounts, σ'', g'', A'', out) ←
+    match c with
+      | ToExecute.Precompiled p =>
+        match p with
+          | 1 =>
+            let (σ', g', A', o) := Ξ_ECREC σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | 2 =>
+            let (σ', g', A', o) := Ξ_SHA256 σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | 3 =>
+            let (σ', g', A', o) := Ξ_RIP160 σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | 4 =>
+            let (σ', g', A', o) := Ξ_ID σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | 5 =>
+            let (σ', g', A', o) := Ξ_EXPMOD σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | 6 =>
+            let (σ', g', A', o) := Ξ_BN_ADD σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | 7 =>
+            let (σ', g', A', o) := Ξ_BN_MUL σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | 8 =>
+            let (σ', g', A', o) := Ξ_SNARKV σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | 9 =>
+            let (σ', g', A', o) := Ξ_BLAKE2_F σ₁ g A I
+            .ok (∅, σ', g', A', some o)
+          | _ => default
```

Comment:
Whoa, what happened here. Please:
```
  let spoon h := .ok (∅, h.1, h.2.1, h.2.2.1, some h.2.2.2)

  -- Equation (131)
  -- Note that the `c` used here is the actual code, not the address. TODO - Handle precompiled contracts.
  let (createdAccounts, σ'', g'', A'', out) ←
    match c with
      | ToExecute.Precompiled p => spoon <|
        match p with
          | 1 => Ξ_ECREC σ₁ g A I
          /- ... -/
          | _ => default
      | ToExecute.Code _ => Ξ debugMode fuel createdAccounts σ₁ g A I
```

Reply:
I honestly took this into consideration but thought having something like `h.2.2.1` is not worthed. Anyway, I made the change